src/cascadia/TerminalSettingsModel/ActionArgs.h

```diff
@@ -125,14 +125,17 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
 
         bool Equals(const Model::NewTerminalArgs& other)
         {
-            return other.Commandline() == _Commandline &&
-                   other.StartingDirectory() == _StartingDirectory &&
-                   other.TabTitle() == _TabTitle &&
-                   other.TabColor() == _TabColor &&
-                   other.ProfileIndex() == _ProfileIndex &&
-                   other.Profile() == _Profile &&
-                   other.SuppressApplicationTitle() == _SuppressApplicationTitle &&
-                   other.ColorScheme() == _ColorScheme;
+            // Treat "default value" as the same as nullopt so that we can
+            // compare two terminal args that don't have all settings set.
+            auto nullEqual = [](auto first, auto& second) { return (first == second) || (first == decltype(first){} && second == std::nullopt); };

```

Comment:
Interesting. It's not immediately apparent to me why this one's needed. I'm assuming it's for the equality check up in `TerminalPage.cpp:388`, but my brain can't figure out why this wouldn't have worked before.

<!-- I wonder - if you flipped the 

```c++
return args.TerminalArgs() == nullptr || args.TerminalArgs().Equals(defaultArgs);
```

statement to 
```c++
return args.TerminalArgs() == nullptr || defaultArgs.Equals(args.TerminalArgs());
```

-->

Reply:
The properties on the autogenerated `NewTerminalArgs` class will return nullptrs and/or `T{}` values if there is no value set, whereas here in the impl we have std::optional<T> values that may or may not `has_value`. So if we try to do a comparison between the impl and the value from the autogenerated property `std::nullopt == nullptr` or `std::nullopt= T{}` will both return false.

Reply:
Made the equals method here more alike all of the others ones: try cast as implementation, then use private variables. I suspect that this code has been actually broken/unused previously.